dataprofiler/tests/profilers/test_profile_builder.py

```diff
@@ -280,15 +329,33 @@ def test_stream_profilers(self, *mocks):
         self.assertEqual(5, profiler.row_has_null_count)
         self.assertEqual(2, profiler.row_is_null_count)
         self.assertEqual(8, profiler.total_samples)
-        self.assertEqual(5, len(profiler.hashed_row_dict))
+        self.assertEqual(5, len(profiler.hashed_row_object))
         self.assertIsNone(profiler.correlation_matrix)
         self.assertDictEqual({"row_stats": 2}, profiler.times)
 
+        # test hll_row_hashing
+        profiler_options = ProfilerOptions()
+        profiler_options.set({"row_statistics.unique_count.hashing_method": "hll"})
+
+        with test_utils.mock_timeit():
+            profiler = dp.StructuredProfiler(data[:3], options=profiler_options)
+
+        self.assertEqual(2, profiler.hashed_row_object.cardinality())
+
+        # check after update
+        with test_utils.mock_timeit():
+            profiler.update_profile(data[3:])
+
+        self.assertEqual(5, profiler.hashed_row_object.cardinality())
+
     def test_correct_unique_row_ratio_test(self):
-        self.assertEqual(2999, len(self.trained_schema.hashed_row_dict))
+        self.assertEqual(2999, len(self.trained_schema.hashed_row_object))
         self.assertEqual(2999, self.trained_schema.total_samples)
         self.assertEqual(1.0, self.trained_schema._get_unique_row_ratio())
 
+        # not completely accurate since register count is 1024
+        self.assertEqual(3046, self.trained_schema_hll.hashed_row_object.cardinality())
```

Comment:
Interesting, this would be concerning to me for the case that the number of duplicate rows would be a negative number.

Reply:
Where is the register count evaluated.